services/components/registry/registry.go

```diff
@@ -69,6 +73,9 @@ func (br *botRegistry) LoadHeartbeatBot() (*config.AgentConfig, error) {
 
 // LoadAssignedBots returns the latest bot list for the running scanner.
 func (br *botRegistry) LoadAssignedBots() ([]config.AgentConfig, error) {
+	br.mu.Lock()
```

Comment:
`GetAgentsIfChanged` can take some time. It's better to wrap `br.botConfigs = agts` line with lock/unlock.